src/components/Detail.vue

```diff
@@ -1,54 +1,63 @@
 <template>
-  <div class="detail">
-    <div class="container">
-      <div class="row bg-secondary mb-4">
-        <div class="col-6">
-          <p>
-            Title
-          </p>
+    <div class="detail">
+        <div class="container">
+            <div class="row bg-secondary mb-4">
+                <div class="col-6">
+                    <p>
+                        Title
+                    </p>
+                </div>
+                <div class="col-3">
+                    Status
+                </div>
+            </div>
+            <div class="row ">
+                <div class="col-6">
+                    <p>
+                        {{ task.title }}
+                    </p>
+                </div>
+                <div class="col-3">
+                    {{ task.isCompleted }}
+                </div>
+            </div>
         </div>
-        <div class="col-3">
-          Status
-        </div>
-      </div>
-      <div class="row ">
-        <div class="col-6">
-          <p>
-            {{ tasks.title }}
-          </p>
-        </div>
-        <div class="col-3">
-          {{ tasks.completed }}
-        </div>
-      </div>
     </div>
-
-    <router-view></router-view>
-  </div>
 </template>
 
-<script>
-import { mapGetters, mapActions } from "vuex";
-export default {
-  name: "Detail",
-  mounted() {
-    console.log(this.$route.params.id);
-    this.getTaskDetails({
-      id: this.$route.params.id,
-    });
-  },
-  beforeUpdate() {
-    console.log(this.tasks);
-  },
-  computed: {
-    ...mapGetters({
-      tasks: "tasks",
-    }),
-  },
-  methods: {
-    ...mapActions(["getTaskDetails"]),
-  },
-};
+<script lang="ts">
+import { Component, Vue, Watch } from 'vue-property-decorator';
+import { Getter, Action } from 'vuex-class';
+import { Task } from '@/types';
+@Component
+export default class Detail extends Vue {
+    @Action('getDetail') readonly getDetail!: Function;
+    @Getter('task') readonly task!: Task;
+    mounted() {
+        console.log(this.$route.params.id);
+        this.getDetail({
```

Comment:
e hạn chể dùng gọi trong mounted và watch để gọi lại `this.getDetail` nha. E có thể research thêm **immediate** của watch nha